WordPress/src/main/java/org/wordpress/android/ui/JetpackRemoteInstallFragment.kt

```diff
@@ -24,48 +22,35 @@ import org.wordpress.android.ui.accounts.LoginActivity
-        super.onCreate(savedInstanceState)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(JetpackRemoteInstallFragmentBinding.bind(view)) {
+            initDagger()
+            initViewModel(savedInstanceState)
+        }
+    }
+
+    private fun initDagger() {
         (requireActivity().application as WordPress).component()!!.inject(this)
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
+    private fun JetpackRemoteInstallFragmentBinding.initViewModel(savedInstanceState: Bundle?) {
         requireActivity().let { activity ->
-            viewModel = ViewModelProvider(this, viewModelFactory).get(JetpackRemoteInstallViewModel::class.java)
-
             val intent = activity.intent
             val site = intent.getSerializableExtra(WordPress.SITE) as SiteModel
             val source = intent.getSerializableExtra(TRACKING_SOURCE_KEY) as JetpackConnectionSource
             val retrievedState = savedInstanceState?.getSerializable(VIEW_STATE) as? JetpackRemoteInstallViewState.Type
+            viewModel = ViewModelProvider(
+                    this@JetpackRemoteInstallFragment, viewModelFactory
+            ).get(JetpackRemoteInstallViewModel::class.java)
             viewModel.start(site, retrievedState)
-            viewModel.liveViewState.observe(viewLifecycleOwner, Observer { viewState ->
-                if (viewState != null) {
-                    if (viewState is JetpackRemoteInstallViewState.Error) {
-                        AppLog.e(AppLog.T.JETPACK_REMOTE_INSTALL, "An error occurred while installing Jetpack")
-                    }
-                    jetpack_install_icon.setImageResource(viewState.icon)
-                    if (viewState.iconTint != null) {
-                        jetpack_install_icon.imageTintList = ColorStateList.valueOf(ContextCompat.getColor(
-                                jetpack_install_icon.context, viewState.iconTint))
-                    } else {
-                        jetpack_install_icon.imageTintList = null
-                    }
-                    jetpack_install_title.setText(viewState.titleResource)
-                    jetpack_install_message.setText(viewState.messageResource)
-                    if (viewState.buttonResource != null) {
-                        jetpack_install_button.visibility = View.VISIBLE
-                        jetpack_install_button.setText(viewState.buttonResource)
-                    } else {
-                        jetpack_install_button.visibility = View.GONE
-                    }
-                    jetpack_install_button.setOnClickListener { viewState.onClick() }
-                    jetpack_install_progress.visibility = if (viewState.progressBarVisible) View.VISIBLE else View.GONE
-                }
-            })
+
+            initLiveViewStateObserver()
```

Comment:
Suggestion (💡): Consider avoiding extra refactoring while converting since this can become harder than necessary to review the changes, even if that is done commit-by-commit. However, since you choose to actually extract that login into the `initLiveViewStateObserver()` private function, why not extracting the below logic as well, and into another `initLiveActionOnResultObserver()` private function. Wdyt?

Reply:
@ParaskP7 - lol, this was the first class I did and came to the same conclusion about refactoring (see last comment and refactoring the parent). The method was too long (detekt), so I pulled out the first observer to pass the rule, knowing that it was simply a lift & move. The second observer was not a lift/move, so I left it.

Reply:
👋 @zwarm !

Thanks for the clarification on that. 🙏